sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2546,15 +2546,37 @@ object functions {
   def soundex(e: Column): Column = withExpr { SoundEx(e.expr) }
 
   /**
-   * Splits str around pattern (pattern is a regular expression).
+   * Splits str around matches of the given regex.
    *
-   * @note Pattern is a string representation of the regular expression.
+   * @param str a string expression to split
+   * @param regex a string representing a regular expression. The regex string should be
+   *              a Java regular expression.
    *
    * @group string_funcs
    * @since 1.5.0
    */
-  def split(str: Column, pattern: String): Column = withExpr {
-    StringSplit(str.expr, lit(pattern).expr)
+  def split(str: Column, regex: String): Column = withExpr {
+    StringSplit(str.expr, Literal(regex), Literal(-1))
+  }
+
+  /**
+   * Splits str around matches of the given regex.
+   *
+   * @param str a string expression to split
+   * @param regex a string representing a regular expression. The regex string should be
+   *              a Java regular expression.
+   * @param limit an integer expression which controls the number of times the regex is applied.
+   *    limit greater than 0: The resulting array's length will not be more than `limit`,
+   *                          and the resulting array's last entry will contain all input beyond
+   *                          the last matched regex.
+   *    limit less than or equal to 0: `regex` will be applied as many times as possible, and
+   *                       the resulting array can be of any size.
```

Comment:
Indentation here looks a bit odd and looks inconsistent at least. Can you double check Scaladoc and format this correctly?

Reply:
will check that -- do you know how I can get around the fact that unidoc throws an error whenever there are logical operators like `<` or `>` in the docs? Having these would be cleaner in general
```
> build/sbt unidoc
...
[error] ./spark/sql/core/target/java/org/apache/spark/sql/functions.java:2834: error: malformed HTML
[error]    *    limit < 0: <code>regex</code> will be applied as many times as possible, and the resulting
[error]               ^